vmss-prototype/vmss-prototype

```diff
@@ -1028,18 +1259,27 @@ def register_commands(subparser):
     parser = add_command(subparser, "status", func=vmss_prototype_status,
                          description='Show status of VMSS Prototype Pattern')
 
-    parser = add_command(subparser, "update", func=vmss_prototype_update,
-                         description='Set up new/updated prototype')
+    parser = add_update_command(subparser, "update", func=vmss_prototype_update,
+                                description='Set up new/updated prototype')
     parser.add_argument('--target-node', required=True, type=NodeName,
                         help='Target node to use as the prototype source')
-    parser.add_argument('--new-updated-nodes', default=0, type=NewUpdatedNodes,
-                        help='Number of new nodes to add to the cluster after the VMSS has been updated from the prototype node (default: %(default)s)')
-    parser.add_argument('--max-history', default=3, type=HistorySize,
-                        help='Number of entries to keep in history - minimum is 2 - (default: %(default)s)')
-    parser.add_argument('--grace-period', type=int, default=300,
-                        help='Grace period in seconds for drain (default: %(default)s)')
-    parser.add_argument('--force', default=False, action='store_true',
-                        help='Force using target node even if drain fails')
+
+    parser = add_update_command(subparser, "auto-update", func=vmss_prototype_auto_update,
+                                description='Automatically set up new/updated prototype if needed')
+    parser.add_argument('--target-vmss', action='append', nargs='+', type=VmssName,
+                        help='Target VMSS pools (all if not given)')
+    parser.add_argument('--pending-reboot-annotation', required=True, type=str,
+                        help='The name of the annotation that, if it exists, the node is pending reboot')
+    parser.add_argument('--last-patch-annotation', required=True, type=str,
+                        help='The name of the annotation that holds the timestamp at which the last patch was applied')
+    parser.add_argument('--node-skip-annotation', action='append', nargs='+', required=False, type=str,
```

Comment:
maybe `--node-ignore-annotation` instead? (very mild opinion here)

Reply:
If we change this to ignore, should the log entries say IGNORE rather than SKIP?

My guess is that they should as it would be more consistent.

Reply:
So, I think having things as IGNORE rather than SKIP is ok - still not a great log line...

Reply:
Now I know what I did not like.  IGNORE is a command IGNORED is what happened.

Ready to merge now?